fsharp-mode-indent-smie.el

```diff
@@ -0,0 +1,95 @@
+;;; fsharp-mode-indent-smie.el --- SMIE indentation for F# -*- lexical-binding: t; coding: utf-8 -*-
+
+;; Copyright (C) 2015 m00nlight Wang
+
+;; Author: 2015 m00nlight Wang <dot.wangyushi@gmail.com>
+
+;; This file is not part of GNU Emacs.
+
+;; This file is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 3, or (at your option)
+;; any later version.
+
+;; This file is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs; see the file COPYING.  If not, write to
+;; the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+;; Boston, MA 02110-1301, USA.
+
+(require 'smie)
+
+
+(defcustom fsharp-indent-level 2
```

Comment:
Change to default 4 space. But since the `fsharp-mode.el` rely on a lot of function defined in `fsharp-mode-indent.el`. I can not easily remove the relevant part of indentation from the code. Just change the default indentation in the mode also to 4.